### ISSUE TITLE
Classification loss function

### ISSUE THREAD

**Issue by deepkshikha** [AlexeyAB/darknet#450]
I have trained tested working fine everything. I want to make some changes in classification loss But I could not understand in which C file classification loss is defined And ?? I found something in Region_layer.c weighted soft max entropy loss kinda please help

**Comment by AlexeyAB** [comment]
* In the `region_layer.c` file there is softmax cross entropy for **Detection:** `-p` and `1-p` (p is probability, i.e. result of softmax): https://github.com/AlexeyAB/darknet/blob/15c89e7a714e7e37c13618eace9325a06f0642fc/src/region_layer.c#L151

----

* In the `cost_layer.c` file there are loss for **Classifier**: https://github.com/AlexeyAB/darknet/blob/15c89e7a714e7e37c13618eace9325a06f0642fc/src/cost_layer.c#L82-L86

    * smooth L1: https://github.com/AlexeyAB/darknet/blob/15c89e7a714e7e37c13618eace9325a06f0642fc/src/blas.c#L220-L235

    * or L2 ([by default for SSE](https://github.com/AlexeyAB/darknet/blob/15c89e7a714e7e37c13618eace9325a06f0642fc/cfg/resnet152.cfg#L1458-L1462) - Sum of Squared Errors): https://github.com/AlexeyAB/darknet/blob/15c89e7a714e7e37c13618eace9325a06f0642fc/src/blas.c#L247-L255

----

If you use GPU and if you want to do some changes in the cost for classification, then you should change these 2 functions: https://github.com/AlexeyAB/darknet/blob/15c89e7a714e7e37c13618eace9325a06f0642fc/src/cost_layer.c#L116-L146

**Comment by deepkshikha** [comment]
Thanks
